hackathon_site/dashboard/frontend/src/api/api.ts

```diff
@@ -50,6 +50,25 @@ export const stripHostname = (url: string): string => {
     return parsed.pathname + parsed.search + parsed.hash;
 };
 
+export const stripHostnameReturnFilters = (
+    url: string | null
+): { path: string; filters?: { [key: string]: any } } => {
+    if (url) {
+        const parsed = new URL(url);
+        const filters: { [key: string]: any } = {};
+        if (parsed.search && filters) {
+            parsed.searchParams.forEach((val, key) => (filters[key] = val));
+        }
+        return JSON.stringify(filters) !== "{}"
```

Comment:
`Object.keys(filters).length !== 0`

Reply:
I thought of doing this but Object.keys() is expensive right because it creates an array?

Reply:
Tbh I'm not sure if it actually creates a new array, or if it is doing an efficient iterator over the keys that are already stored. That being said, `Object.keys(obj).length === 0` is the most common way of checking for an empty object that I've seen.